src/libs/auth.js

```diff
@@ -82,19 +82,30 @@ const revokeTokens = async () => {
 }
 
 const getSigninArgs = includeBillingScope => {
-  return includeBillingScope === true ? {
-    scope: `
-    openid
-    email
-    profile
-    https://www.googleapis.com/auth/cloud-billing
-  `.trim().split(/\s+/).join(' ')
-  } : {}
+  return Utils.cond(
+    [includeBillingScope === false, () => {}],
+    // For Google just append the scope to the signin args.
+    [isGoogleAuthority(), () => ({ scope: 'openid email profile https://www.googleapis.com/auth/cloud-billing' })],
+    // For B2C switch to a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    () => ({
```

Comment:
@nmalfroy not sure if TDR is doing this yet -- but fyi I was able to dynamically switch the B2C policy by setting these params on the login request.

Reply:
Cool.  We don't do any scope switching but this might be useful to do in the near future

Reply:
that's interesting...is the `p` parameter a documented one or did you have to figure that one out

Reply:
`p` is the B2C policy (e.g. https://terradevb2c.b2clogin.com/terradevb2c.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=b2c_1a_signup_signin_billing)

These options just append extra querystring params to the authorize/token endpoints. (I also updated the backend routes to handle incoming requests with `p` correctly.)